src/Compilers/CSharp/Test/Syntax/Parsing/LambdaAttributeParsingTests.cs

```diff
@@ -0,0 +1,1777 @@
+                            {
+                                N(SyntaxKind.IdentifierToken, "x");
+                            }
+                            N(SyntaxKind.EqualsGreaterThanToken);
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "y");
+                            }
+                        }
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+            }
+            EOF();
+        }
+
+        // Lambda with attribute in collection initializer is similar to dictionary initializer.
+        [Fact]
+        public void CollectionInitializer_02()
+        {
+            UsingExpression("new B { ([A] x => y) }", TestOptions.RegularPreview);
+
+            N(SyntaxKind.ObjectCreationExpression);
+            {
+                N(SyntaxKind.NewKeyword);
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "B");
+                }
+                N(SyntaxKind.CollectionInitializerExpression);
+                {
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.ParenthesizedExpression);
+                    {
+                        N(SyntaxKind.OpenParenToken);
+                        N(SyntaxKind.SimpleLambdaExpression);
+                        {
+                            N(SyntaxKind.AttributeList);
+                            {
+                                N(SyntaxKind.OpenBracketToken);
+                                N(SyntaxKind.Attribute);
+                                {
+                                    N(SyntaxKind.IdentifierName);
+                                    {
+                                        N(SyntaxKind.IdentifierToken, "A");
+                                    }
+                                }
+                                N(SyntaxKind.CloseBracketToken);
+                            }
+                            N(SyntaxKind.Parameter);
+                            {
+                                N(SyntaxKind.IdentifierToken, "x");
+                            }
+                            N(SyntaxKind.EqualsGreaterThanToken);
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "y");
+                            }
+                        }
+                        N(SyntaxKind.CloseParenToken);
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+            }
+            EOF();
+        }
+
+        // Lambda with attribute in conditional expression is similar to conditional element access.
+        [Fact]
+        public void ConditionalExpression()
+        {
+            UsingExpression("x ? ([A] () => { }) : y", TestOptions.RegularPreview);
```

Comment:
Consider testing without the parens around the lambda.

Reply:
Or consider moving this to the location below where you do test it to put like things together :)

---
In reply to: [600890034](https://github.com/dotnet/roslyn/pull/52038#discussion_r600890034) [](ancestors = 600890034)